src/app/containers/AssignmentDutyDropTarget.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react'
+import { CSSProperties } from 'react';
+import { 
+    default as dropTargetFactory, 
+    DragDropStatus 
+} from '../infrastructure/DragDrop/dropTargetFactory';
+import ItemTypes from '../infrastructure/DragDrop/ItemTypes';
+import { AssignmentDuty, Sheriff } from '../api';
+
+type DroppableItem = Sheriff;
+
+const GenericAssignmentDropTarget = dropTargetFactory<DroppableItem, void>(ItemTypes.SHERIFF);
+
+interface AssignmentDutyDropTargetProps {
+    duty: AssignmentDuty;
+    canDropItem?: (item: DroppableItem) => boolean;
+    onDropItem: (item: DroppableItem) => void;
+    style?: CSSProperties;
+    computeStyle?: (status: { isActive: boolean, isOver: boolean, canDrop: boolean }) => CSSProperties;
+}
+
+export default class AssignmentDutyDropTarget extends React.PureComponent<AssignmentDutyDropTargetProps> {
+
+    static defaultProps: Partial<AssignmentDutyDropTargetProps> = {
+        canDropItem: (i: DroppableItem) => true
+    }
+
+    // canDropItem(sheriff: DraggedSheriff) {
```

Comment:
Can this be removed?

Reply:
Actually didn't need this class anymore anyways, so 👋